public/app/features/alerting/EditNGAlert.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { MapDispatchToProps, MapStateToProps } from 'react-redux';
+import { Spinner } from '@grafana/ui';
+import { connectWithCleanUp } from '../../core/components/connectWithCleanUp';
+import NextGenAlertingPage from './NextGenAlertingPage';
+import { getAlertDefinition, updateAlertDefinition } from './state/actions';
+import { getRouteParamsId } from '../../core/selectors/location';
+import { AlertDefinition, StoreState } from '../../types';
+
+interface OwnProps {}
+
+interface ConnectedProps {
+  alertDefinition: AlertDefinition;
+  pageId: string;
+}
+
+interface DispatchProps {
+  getAlertDefinition: typeof getAlertDefinition;
+  updateAlertDefinition: typeof updateAlertDefinition;
+}
+
+type Props = ConnectedProps & DispatchProps & OwnProps;
+
+export class EditNGAlert extends PureComponent<Props> {
+  componentDidMount() {
+    const { pageId, getAlertDefinition } = this.props;
+    getAlertDefinition(pageId);
+  }
+
+  render() {
+    const { alertDefinition, updateAlertDefinition } = this.props;
+
+    if (alertDefinition.id < 1) {
+      return <Spinner />;
+    }
+    return <NextGenAlertingPage alertDefinition={alertDefinition} saveDefinition={updateAlertDefinition} />;
+  }
+}
+
+const mapStateToProps: MapStateToProps<ConnectedProps, OwnProps, StoreState> = (state) => {
+  const pageId = getRouteParamsId(state.location);
+  return {
+    alertDefinition: state.alertDefinition.alertDefinition,
+    pageId: (pageId as string) ?? '',
+  };
+};
+
+const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = {
+  getAlertDefinition,
+  updateAlertDefinition,
+};
+
+export default hot(module)(
+  connectWithCleanUp(mapStateToProps, mapDispatchToProps, (state) => state.alertDefinition)(EditNGAlert)
+);
```

Comment:
What is the reason why we use PureComponent instead of Functional Component?

Reply:
I want these to connect to the store and while I think that's kind of possible in functional components too I think  it looks cleaner in a PureComponent, is there a huge upside to using functional components here?

Reply:
I was just curious really, seems like such a small Component so it felt like it should be a Functional Component and all the ceremony of connecting to Redux is much less in a Functional Component.

Reply:
Yeah, maybe it's a bit overkill.

Reply:
Problem solved :slightly_smiling_face: